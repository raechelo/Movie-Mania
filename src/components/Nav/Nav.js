import React from 'react';
import lamp from '../../assets/Pixar-Lamp.svg';
import { NavLink } from 'react-router-dom';

const Nav = () => {
  return (
    <header className="Header">
      <div>
        <img src={lamp} />
        <h1>Movie Mania</h1>
      </div>
      <nav>
        <NavLink to='/' className='nav'>Home</NavLink>
        <NavLink to='/favorites' className='nav'>Favorites</NavLink>
        <NavLink to='/login' className='nav'>Login</NavLink>
        <NavLink to='/sign-up' className='nav'>Sign Up</NavLink>
      </nav>
    </header>
  )
}

export default Nav