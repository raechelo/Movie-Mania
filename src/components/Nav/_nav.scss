.Header {
  h1 {
    display: inline;
    grid-area: title;
    font-family: $fancyfont;
    font-size: 3rem;
    margin: auto;
  }

  nav {
    color: darken($accent, 30%);
    font-family: $fancyfont;
    font-size: 1.5rem;
    display: grid;
    grid-template-areas: ". title login";
    grid-template-columns: repeat(3, 33%);

    h3 {
      color: $accent;
      text-shadow: 1px 1px 1px lighten($accent, 70%);
      margin: 0;
    }

    a {
      color: white;
      border: 3px solid transparent;
      padding: 10px;

      &:hover {
        color: $accent;
      }
    }

    .active {
      color: $accent;
    }
  }

  img {
    width: 2.5rem;
  }
  
  div {
    grid-area: login;
    margin-top: 1.5rem;
    text-align: center;
  }
}

.title {
  grid-area: title;
  margin: auto;
}