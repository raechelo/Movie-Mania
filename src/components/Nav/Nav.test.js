import React from 'react';
import { shallow } from 'enzyme';
import { Nav } from './Nav';

describe('Nav', () => {

  let wrapper;

  let user = { name: 'Zeus' }

  beforeEach(() => {
    wrapper = shallow( <Nav user={user} /> )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  });
  
});