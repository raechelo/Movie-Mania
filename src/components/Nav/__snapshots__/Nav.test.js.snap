// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot with all the data passed in 1`] = `
<header
  className="Header"
>
  <nav>
    <NavLink
      className="nav title"
      exact={true}
      to="/"
    >
      <h1>
        movie mania
      </h1>
    </NavLink>
    <div>
      <NavLink
        className="nav"
        exact={true}
        to="/login"
      >
        <i
          className="fas fa-skull"
        />
        Login
      </NavLink>
      <NavLink
        className="nav"
        exact={true}
        to="/sign-up"
      >
        <i
          className="fas fa-book-dead"
        />
        Sign Up
      </NavLink>
    </div>
  </nav>
</header>
`;
